library/src/main/java/io/github/boguszpawlowski/composecalendar/states/CurrentState.kt

```diff
@@ -0,0 +1,38 @@
+package io.github.boguszpawlowski.composecalendar.states
+
+import androidx.compose.runtime.Stable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.saveable.Saver
+import androidx.compose.runtime.setValue
+import java.time.LocalDate
+
+@Suppress("FunctionName") // Factory function
+public fun CurrentState(initialDay: LocalDate): CurrentState = CurrentStateImpl(initialDay)
```

Comment:
Also, I believe having the necessity to create such `CurrentState` is a side-effect of handling both `Month` and `Week` mode in the same composable. IMO this should be 2 separate composables - `Calendar` and `WeekCalendar`. This would allow us to avoid unnecessary complication of the code.